types/react-native/index.d.ts

```diff
@@ -962,6 +962,12 @@ export interface TextPropsAndroid {
      * default is `highQuality`.
      */
     textBreakStrategy?: 'simple' | 'highQuality' | 'balanced';
+
+    /**
+     * Determines the types of data converted to clickable URLs in the text element.
+     * By default no data types are detected.
+     */
+    dataDetectorType?: 'phoneNumber' | 'link' | 'email' | 'none' | 'all';
```

Comment:
As the type declaration in the react native source code (https://github.com/facebook/react-native/blob/49015b0f5bda83794b88b17dd3cbd834fa235b72/Libraries/Text/TextProps.js#L175) is a Flow maybe (https://flow.org/en/docs/types/maybe/), this type here should include `null` :)
Can we add a test in addition?

Reply:
Thanks for the review 👍🏻 
Good catch with nullable 👏🏻  

As i'm not an expert with TS and react native ecosystem, i have some remarks : 
* ❌ Use `null` value in `dataDetectorType` props throw an error (screenshot below, i try it in my own app) 
![Screenshot_20201005_120344](https://user-images.githubusercontent.com/22751731/95067022-7c802000-0703-11eb-8d0d-9a712839f13e.png)
* ✅ Its seems to be nullable in the [java code](https://github.com/facebook/react-native/blob/0.63-stable/ReactAndroid/src/main/java/com/facebook/react/views/text/ReactTextAnchorViewManager.java#L167-L185)
* ❓ `dataDetectorTypes` (for TextInput) [seems to not allow null by default](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/991cb258984ea975debcfa1bf2dac5bbb6829ce2/types/react-native/index.d.ts#L1070) in this repo

So, as i'm not comfortable with the environment, feel free to tell me if i'm wrong and i'll update the PR for this part :) 

About the test, i tried to take a look at the tests but they seems "stuffy", how it should works ? i only see react components with no call, [example](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/991cb258984ea975debcfa1bf2dac5bbb6829ce2/types/react-native/test/index.tsx#L1463), so how the CI or npm test can really test the component ? If you have more info, documentation or whatever, it would very help me 🙏🏻

Reply:
I was thinking about unit or functional tests but it seems its only type check declaration, let me know :)

Reply:
@ca057 Is correct that the upstream Flow typing is a ‘maybe’ type, which means it _should_ accept a `null` value. So it would appear that either the implementation has a bug and should accept `null` _or_ the type shouldn’t actually be a ‘maybe’ type. I’m inclined to believe it’s the former. So let’s type it as accepting `null` and then it would be great if you could send a PR upstream to also make it accept `null` besides just `undefined`.

Regarding testing, yes these tests are not ran, it’s _only_ about checking that normal API usage does not lead to type errors. _Or_, where necessary, you can assert that a type error is raised.